R/deploy-site.R

```diff
@@ -98,52 +97,50 @@ deploy_site_github <- function(
   cat_line("Setting private key permissions to 0600")
   fs::file_chmod(ssh_id_file, "0600")
 
-  deploy_local(pkg, repo_slug = repo_slug, host = host, commit_message = commit_message, ...)
+  deploy_to_branch(pkg, commit_message = commit_message, branch = "gh-pages", ...)
 
   rule("Deploy completed", line = 2)
 }
 
-deploy_local <- function(
-                         pkg = ".",
-                         repo_slug = NULL,
-                         host,
+#' Build and deploy a site locally
+#'
+#' @param branch The git branch to deploy to
+#' @param ... Additional arguments passed to [build_site()].
+#' @inheritParams build_site
+#' @inheritParams deploy_site_github
+#' @export
+deploy_to_branch <- function(pkg = ".",
                          commit_message = construct_commit_message(pkg),
-                         ...
-                         ) {
+                         branch = "gh-pages",
+                         ...) {
   dest_dir <- fs::dir_create(fs::file_temp())
   on.exit(fs::dir_delete(dest_dir))
 
-  pkg <- as_pkgdown(pkg)
-  if (is.null(repo_slug)) {
-    gh <- rematch2::re_match(pkg$github_url, github_url_rx())
-    repo_slug <- paste0(gh$owner, "/", gh$repo)
-  }
-
-  github_clone(dest_dir, repo_slug, host)
+  git("fetch", "origin", branch)
+  github_worktree_add(dest_dir)
```

Comment:
I may be completely wrong here but shouldn't this be:

```r
github_worktree_add(dest_dir, branch = branch)
```

?

We're getting an error when trying to deploy from travis with the dev version of pkgdown version:

> Error in git("worktree", "add", "--track", "-b", branch, dir, paste0("origin/",  : 
>
>  argument "branch" is missing, with no default

Complete log: https://travis-ci.org/rmaia/pavo/jobs/651466810

Reply:
You are right @Bisaloo, should be fixed by https://github.com/r-lib/pkgdown/commit/aceba8b49f4d20fcb9ca11549b6c477143aca5f4

Reply:
This change now throws the following error when travis tries to deploy on enterprise github


> Running git worktree add --track -b gh-pages /tmp/Rtmp8zQQdq/file598365058050 \
  origin/gh-pages
Preparing worktree (new branch 'gh-pages')
error: the requested upstream branch 'origin/gh-pages' does not exist
...
...
...
Error: System command 'git' failed, exit status: 255, stdout & stderr were printed